File: java-car-data/car-data-web-ui-tests/src/test/java/LoginPage.java
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class LoginPage {

    @FindBy(id = "username")
    private WebElement userNameInput;

    @FindBy(id = "password")
    private WebElement passwordInput;

    @FindBy(xpath = "//form[@id='loginform']//input[@type='submit']")
    private WebElement signInButton;

    public LoginPage(WebDriver driver) {
        PageFactory.initElements(driver, this);
    }

    public void typeUserName(String userName) {
        userNameInput.sendKeys(userName);
    }

    public void typePassword(String password) {
        passwordInput.sendKeys(password);
    }

    public void clickSignInButton() {
        signInButton.click();
    }

    public void login(String username, String password) {
        typeUserName("user");
        typePassword("password");
        clickSignInButton();
    }
}

File: python-selenium/login_page.py
from locators import LoginPageLocators
from page import BasePage

class LoginPage(BasePage):
    def login(self, username, password):
        username_input = self.driver.find_element(*LoginPageLocators.USERNAME_INPUT)
        password_input = self.driver.find_element(*LoginPageLocators.PASSWORD_INPUT)
        sign_in_button = self.driver.find_element(*LoginPageLocators.SIGN_IN_BUTTON)

        username_input.send_keys("user")
        password_input.send_keys("password")
        sign_in_button.click()
